experimental/Schemes/Sheaves.jl

```diff
@@ -0,0 +1,106 @@
+export AbsSheaf
+export space, restriction_map
+export SheafOnScheme
+export RingOfRegularFunctions
+
+########################################################################
+# The AbsSheaf interface                                               #
+########################################################################
+@Markdown.doc """
+    space(F::AbsSheaf) 
+
+For a sheaf ``ℱ`` on a space ``X`` return ``X``.
+"""
+function space(F::AbsSheaf) 
+  return space(underlying_sheaf(F))
+end
+
+@Markdown.doc """
+    (F::AbsSheaf)(U; cached=true) 
+
+For a sheaf ``ℱ`` on a space ``X`` and an (admissible) open set 
+``U ⊂ X`` check whether ``U`` is open in ``X`` and return ``ℱ(U)``.
+"""
+function (F::AbsSheaf{<:Any, OpenType, OutputType})(U::T; cached::Bool=true) where {OpenType, OutputType, T<:OpenType}
+  return (underlying_sheaf(F))(U; cached=cached)::OutputType
+end
+
+@Markdown.doc """
+    restriction_map(F::AbsSheaf{<:Any, OpenType, OutputType},
+                    U::Type1, V::Type2
+                   ) where {
+                     OpenType, OutputType, 
+                     Type1<:OpenType, Type2<:OpenType
+                   }
+
+For a sheaf ``ℱ`` on a space ``X`` and an (admissible) pair of 
+open sets ``U, V ⊂ X`` check whether ``U ⊂ V ⊂ X`` are open and 
+return the restriction map ``ℱ(V) → ℱ(U)``.
+"""
+function restriction_map(F::AbsSheaf{<:Any, OpenType, OutputType, RestrictionType},
+    U::Type1, V::Type2
+  ) where {OpenType, OutputType, RestrictionType, Type1<:OpenType, Type2<:OpenType}
+  return restriction_map(underlying_sheaf(F), U, V)::RestrictionType
+end
+
+# An alias for shorter notation
+function (F::AbsSheaf{<:Any, OpenType, OutputType, RestrictionType})(
+    U::Type1, V::Type2
+  ) where {OpenType, OutputType, RestrictionType, Type1<:OpenType, Type2<:OpenType}
+  return restriction_map(F, U, V)::RestrictionType
+end
+
+@Markdown.doc """
+    is_open_func(F::AbsSheaf)
+
+For a sheaf ``ℱ`` on a space ``X`` return a function `f` on two 
+arguments such that `f(U, V)` returns `true` whenever ``U ⊂ V ⊂ X``
+are open inclusions and `false` otherwise.
+
+***Note:*** This function must accept ``X`` as the second argument 
+to check whether ``U ⊂ X`` is open in ``X``!
+"""
```

Comment:
Maybe change `must` to `will`? All the docstrings are written in a way explaining what the function will do ("return ...") and never what the function must do.